registry/client/transport/http_reader.go

```diff
@@ -46,16 +50,23 @@ func (hrs *httpReadSeeker) Read(p []byte) (n int, err error) {
 		return 0, hrs.err
 	}
 
+	if hrs.offset != hrs.seekOffset {
```

Comment:
Can this statement be in `(*httpReadSeeker).Seek`? I think we can avoid having both `seekOffset` and `offset`.